cmake/SetupCodeMetrics.cmake

```diff
@@ -49,7 +49,28 @@
 # Note: Only supported for gnu or clang.
 ########################################
 if (ENABLE_COVERAGE)
-    include(blt/cmake/CodeCoverage.cmake)
+    ##########################################################################
+    # Setup coverage compiler flags 
+    ##########################################################################
+    # Set the actual flags for coverage in the COVERAGE_FLAGS variable 
+    # Note: '--coverage' is equivalent to '-fprofile-arcs -ftest-coverage' for compiling and '-lgcov' for linking
+    # Additional flags that might be useful: 
+    #       " -fno-inline -fno-inline-small-functions -fno-default-inline"
+    blt_append_custom_compiler_flag(FLAGS_VAR   COVERAGE_FLAGS 
+                                    DEFAULT     "--coverage")
```

Comment:
I was suggesting that we only set the flag on the compilers that we know are valid, rather than setting it for all compilers.  We can add the proper flags when we need them.